src/loop_in_chunks.cpp

```diff
@@ -520,4 +530,69 @@ void fields::loop_in_chunks(field_chunkloop chunkloop, void *chunkloop_data,
   }
 }
 
+/***************************************************************/
+/***************************************************************/
+/***************************************************************/
+std::vector<double> fields::get_array_metadata(const volume &where)
+{
+  component cgrid=Centered;
+  vec yee_c(gv.yee_shift(Centered) - gv.yee_shift(cgrid));
+  ivec iyee_c(gv.iyee_shift(Centered) - gv.iyee_shift(cgrid));
+  volume wherec(where + yee_c);
+  ivec is(vec2diel_floor(wherec.get_min_corner(), gv.a, zero_ivec(gv.dim)));
+  ivec ie(vec2diel_ceil(wherec.get_max_corner(), gv.a, zero_ivec(gv.dim)));
+  
+  ivec imin=gv.little_corner()+one_ivec(gv.dim), imax=gv.big_corner()-one_ivec(gv.dim);
+  LOOP_OVER_DIRECTIONS(gv.dim, d)
+   { if (is.in_direction(d) < imin.in_direction(d))
+      is.set_direction(d,imin.in_direction(d));
+     if (ie.in_direction(d) > imax.in_direction(d))
+      ie.set_direction(d,imax.in_direction(d));
+   }
+
+  bool snap_empty_dims=true;
+  vec s0(gv.dim), e0(gv.dim), s1(gv.dim), e1(gv.dim);
+  // this initialization step seems to be necessary here to
+  // avoid winding up with zero or undefined integration
+  // weights; I don't know why it seems to be unnecessary
+  // for loop_in_chunks above.
+  FOR_DIRECTIONS(d)
+   if (!has_direction(gv.dim,d))
+    { s0.set_direction(d,1.0);
+      e0.set_direction(d,1.0);
+      s1.set_direction(d,1.0);
+      e1.set_direction(d,1.0);
+    }
+  compute_boundary_weights(gv, wherec, is, ie, snap_empty_dims,
+                           s0, e0, s1, e1);
+
+  // Determine integration "volumes" dV0 and dV1;
+  double dV0 = 1.0, dV1 = 0.0;
+  LOOP_OVER_DIRECTIONS(gv.dim, d)
+   if (wherec.in_direction(d) > 0.0)
+    dV0 *= gv.inva;
+
+  // punt on figuring out what to do here
+  if (gv.dim == Dcyl)
+   fprintf(stderr,"** warning: cylindrical coordinates not supported in get_array_metadata; integration weights may be incorrect\n");
+
+  size_t array_size=1;
+  LOOP_OVER_DIRECTIONS(gv.dim, d)
+   array_size *= (1 + iabs(ie.in_direction(d)-is.in_direction(d))/2);
+  std::vector<double> xyzw(4*array_size,0.0);
+  int np=0;
+  LOOP_OVER_IVECS(gv, is, ie, idx)
+   { 
+     IVEC_LOOP_LOC(gv, loc);
+     double w = IVEC_LOOP_WEIGHT(s0, s1, e0, e1, dV0 + dV1 * loop_i2);
+
+     if (has_direction(gv.dim,X)) xyzw[4*np+0] = loc.x();
```

Comment:
Instead, if you allocate a 1d `x` array, do:
```c
IVEC_LOOP_ILOC(gv, iloc);

if (has_direction(gv.dim,X)) xyzw[(iloc.in_direction(X) - is.in_direction(X))/2] = loc.x();
```
(This overwrites the `x` array multiple times, but computational cost is irrelevant here.)